File: src/main/java/me/creepinson/creepinoutils/util/IConnectable.java
package me.creepinson.creepinoutils.util;

import net.minecraft.util.Direction;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IWorld;

/**
 * @author Theo Paris https://theoparis.com
 * Project creepinoutils
 **/
public interface IConnectable {
    boolean canConnectTo(IWorld blockAccess, BlockPos pos, Direction side);

    boolean canConnectToStrict(IWorld blockAccess, BlockPos pos, Direction side);
}

File: src/main/kotlin/com/theoparis/creepinoutils/util/api/IConnectable.kt
package com.theoparis.creepinoutils.util.api

import net.minecraft.util.Direction
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IWorld

/**
 * @author Theo Paris https://theoparis.com
 * Project creepinoutils
 */
interface IConnectable {
    fun canConnectTo(blockAccess: IWorld, pos: BlockPos, side: Direction?): Boolean
    fun canConnectToStrict(blockAccess: IWorld, pos: BlockPos, side: Direction?): Boolean
}